src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -346,16 +348,38 @@ protected boolean isSnapshotRequired(GetCheckpointResponse getCheckpointResponse
 
         return false;
       } else {
-        // Set checkpoint with bootstrap and initialCheckpoint as false.
-        // A call to set the checkpoint is required first otherwise we will get an error 
-        // from the server side saying:
-        // INTERNAL_ERROR[code 21]: Stream ID {} is expired for Tablet ID {}
-        makeStreamActive(tableId, tabletId, false);
+        // Mark the stream as active by setting a checkpoint.
+        if (!OpId.isValid(getCheckpointResponse.getTerm(), getCheckpointResponse.getIndex())) {
+          // Set checkpoint with bootstrap as false and initialCheckpoint as true.
+          // A call to set the checkpoint is required first otherwise we will get an error 
+          // from the server side saying:
+          // INTERNAL_ERROR[code 21]: Stream ID {} is expired for Tablet ID {}
+          makeStreamActive(tableId, tabletId, false);
+        }
 
         return true;
       }
     }
 
+    /**
+     * This method will ONLY be called when we are sure that we have to take the snapshot.
+     * @param getCheckpointResponse
+     * @return the from_op_id to start snapshot with
+     */
+    protected OpId getSnapshotStartLsn(GetCheckpointResponse getCheckpointResponse) {
+      Objects.requireNonNull(getCheckpointResponse);
+
+      if (getCheckpointResponse.getTerm() > 0 && getCheckpointResponse.getIndex() > 0) {
```

Comment:
Can the came condition of valid lsn be used here? Or do we see any issue if OpId is (0,0)? We have to be careful.

Reply:
It can be `0.0` where the connector just failed after setting the checkpoint for the first time i.e. marking the stream as active - but even in that case, we only want the snapshot to be requested from `-1.-1` as this is still the beginning.